pinot-spi/src/main/java/org/apache/pinot/spi/stream/StreamPartitionMsgOffset.java

```diff
@@ -40,7 +40,7 @@
  * versions of the stream implementation
  */
 @InterfaceStability.Evolving
-public interface StreamPartitionMsgOffset extends Comparable {
+public interface StreamPartitionMsgOffset extends Comparable<StreamPartitionMsgOffset> {
```

Comment:
This caused a lot of warnings which make code harder to read, and helps the compiler to prevent heap pollution bugs.